src/DatePicker/DatePickerInputOverlay.tsx

```diff
@@ -0,0 +1,22 @@
+import { DayPickerProps } from 'react-day-picker';
+
+import Box from '../Box';
+import DatePicker from './DatePicker';
+import PopoverContent, {
+  Props as PopoverContentProps
+} from '../Popover/PopoverContent';
+
+type Props = PopoverContentProps & {
+  children: { props: DayPickerProps };
```

Comment:
We actually only care about `children`'s props here, which are given to us by `react-day-picker`; we pass this component as a custom `overlayComponent` to `DayPickerInput`.